main.tf

```
@@ -39,7 +39,7 @@ resource "aws_security_group_rule" "ingress_cidr" {
 }
 
 resource "aws_security_group_rule" "ingress_sg" {
-  count = local.create_ingress_cidr_sg_rule ? length(var.security_group_ids) : 0
+  count = local.create_ingress_sg_sg_rule ? length(var.security_group_ids) : 0
```

This is better than what you had, but you should consider changing this from `count` to:
```suggestion
  for_each = toset(
    local.create_ingress_sg_sg_rule ?
    var.security_group_ids :
    []
  )
```

---

`count` can run into issues with the count not being available until apply time. `for_each` doesn't have those issues.

You will need additional changes to accommodate the fact that it's now a set instead of a list, and `count.index` isn't the correct way to access a specific group.

---

Thanks, I'll use `for_each` instead of count in the future but I've got more to do this week than I have time for